apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -118,6 +120,16 @@ def __init__(
                 provided, a new ThreadPoolExecutor will be created with
                 default settings.
         """
+        if tiledb_ctx is not None:
+            version = get_implementation_version().split(".")
+            assert (int(version[0]), int(version[1])) < (1, 14)
```

Comment:
It seems like it would make sense to have a function that returns this as a tuple or a `packaging.version.Version`.